### ISSUE TITLE
Swagger and wire values don't match leading to failed deserialization

### ISSUE THREAD

**Issue by jhendrixMSFT** [Azure/azure-rest-api-specs#7003]
Ported from https://github.com/Azure/azure-sdk-for-go/issues/5549

See reference [Path](https://github.com/Azure/azure-rest-api-specs/blob/master/specification/storage/data-plane/Microsoft.StorageDataLake/stable/2018-11-09/DataLakeStorage.json#L88).

The swagger defines `isDirectory` as `boolean` however the value sent over-the-wire is the string "true".
The swagger defines `contentLength` as `integer` however the value sent over-the-wire is the string "123".

**Comment by kahing** [comment]
Note that the deserialization works fine in Java (for example), so it's not clear where you want to fix this

**Comment by jhendrixMSFT** [comment]
Yes, some languages have much more relaxed deserializers, Java and C# being the canonical examples.  Go's deserializer is much more strict so the swagger must match what's on the wire, and generally speaking the swagger should always match what's on the wire for documentation purposes.

**Comment by kahing** [comment]
It'd be nice if azure-sdk-go uses a custom demarshaller instead of returning string for those fields

**Comment by jhendrixMSFT** [comment]
Agreed that returning the values as strings is sub-optimal.  It's likely a bug in the service itself although fixing it there can be more difficult.
Regarding a custom unmarshaller, it's something I've considered but is a large work item.

**Comment by sumantmehtams** [comment]
Hi @kahing 

ADLS Gen1 is mostly in maintenance mode. Bug fixes are ok but any breaking changes involving swagger updates or REST endpoint changes is something we're avoiding. 

Will close this issue but feel free to reopen.
